### PR TITLE
feat(workflow): generate transcript with screenshots from video

### PR DESCRIPTION
**Pull request by vojtaholik**
data preview screenshot
inngest function screenshot

capturing this for later gif

### CODE REVIEW

File: apps/course-builder-web/src/inngest/functions/ai/data/generate-transcript-with-screnshots.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+      throw new NonRetriableError(`Video resource not found for id (${event.data.videoResourceId})`)
+    }
+    const transcriptWithScreenshots = await step.run('generate transcript with screenshots', async () => {
+      return await mergeSrtWithScreenshots(videoResource.srt, videoResource.muxPlaybackId)
+    })
+    return { transcriptWithScreenshots }
+  },
+)
+
+async function mergeSrtWithScreenshots(srt: string, muxPlaybackId: string) {
```
Comment (vojtaholik) [inline apps/course-builder-web/src/inngest/functions/ai/data/generate-transcript-with-screnshots.ts:30]: this is based on @tayiorbeii’s [screenshot-from-srt-args](https://github.com/skillrecordings/transcript-automation-kenv/blob/main/scripts/screenshot-srt-from-args.js) script

File: apps/course-builder-web/src/lib/resources.ts [hunk 2]
```diff
@@ -9,5 +9,8 @@ export async function getResource<T = any>(slugOrId: string): Promise<T | null>
           summary,
           body,
           "slug": slug.current,
+          srt,
+          wordLevelSrt,
+          muxPlaybackId,
   }`)
```
Comment (joelhooks) [inline apps/course-builder-web/src/lib/resources.ts:15]: something to keep in mind/recognize here is that `getResource` was **type agnostic** but this wholly makes this `getVideoResource` and should be typed and moved into it's own lib

File: apps/course-builder-web/src/trpc/api/routers/tips.ts [hunk 3]
```diff
@@ -145,4 +145,24 @@ export const tipsRouter = createTRPCRouter({
...
+    )
+    .mutation(async ({ ctx, input }) => {
+      const session = await getServerAuthSession()
+      const ability = getAbility({ user: session?.user })
+      if (!ability.can('create', 'Content')) {
+        throw new Error('Unauthorized')
+      }
+
+      return await inngest.send({
+        name: TRANSCRIPT_WITH_SCREENSHOTS_REQUESTED_EVENT,
```
Comment (joelhooks) [inline apps/course-builder-web/src/trpc/api/routers/tips.ts:162]: with Inngest, and more broadly with event driven workflows, you want to tell the system what has happened, and not request specific work

imo, this should simply occur with `VIDEO_UPLOADED` or `TRANSCRIPT_COMPLETE` and not as a remote procedure call (rpc via tRPC)

File: apps/course-builder-web/src/inngest/functions/ai/data/generate-transcript-with-screnshots.ts (hunk 1 above)
Comment (joelhooks) [reply apps/course-builder-web/src/inngest/functions/ai/data/generate-transcript-with-screnshots.ts:30]: this needs to be extracted into a `transcript-processing` folder that we could move other similar utils in to and create a package from

**Review by joelhooks — APPROVED** [review]

**Pull request merged**